### PR TITLE
Extend the history command with a "transfer" subcommand

### PR DESCRIPTION
**Pull request by cryzed**
Extend the history command with a "transfer" subcommand to transfer data between history backends

Extended history backend interface's `all_items()` method with a `full_item: bool` keyword-argument, that would yield a tuple of sessionid and the full "cmd"-dict.

Sample usage:

- Transfers the dummy entry into a new JSON history session: `xonsh history transfer --source dummy json`
- Transfers all JSON history entries into the dummy: `xonsh history transfer --source json dummy`
- Transfers from the configured default source (`$XONSH_HISTORY_BACKEND` or "json") to the target "sqlite": `xonsh history transfer sqlite`

`--source-filename` and `--target-filename` are available to potentially override the default value passed into the `filename` keyword-argument for the History implementation, in case the user has specified non-standard log locations.

There are no test for this currently, nor have I adjusted any of the existing tests for the `full_items()` change to the History interface yet. I would like to hear your thoughts on this first and maybe tips on what tests to touch. When and if everything looks ok, I will also add the news entry.

Original discussion here: https://gitter.im/xonsh/xonsh?at=5f79f6446e0eb8446965538b

### CODE REVIEW

File: xonsh/history/dummy.py [hunk 1]
```diff
@@ -11,10 +11,15 @@ def append(self, cmd):
...
 
     def items(self, newest_first=False):
-        yield {"inp": "dummy in action", "ts": 1464652800, "ind": 0}
+        yield {"inp": "dummy in action", "ts": 1464652800, "ind": 0, "rtn": 0}
 
-    def all_items(self, newest_first=False):
-        return self.items(newest_first=newest_first)
+    def all_items(self, newest_first=False, full_item=False):
+        if full_item:
+            sessionid = self.sessionid or uuid.uuid4()
```
Comment (gforsyth) [inline xonsh/history/dummy.py:18]: `uuid` isn't imported, which is causing the `flake` error

**Comment by gforsyth** [comment]
Thanks for putting this in @cryzed !  This looks pretty cool to me.  We'll want to have some tests to make sure it's working as expected.

I can also envision (not for this PR, just forward looking) having a markdown or an org-mode "backend" that would be super handy for writing out documentation / notes with example commands + their outputs

**Comment by daniel-shimon** [comment]
Hi @cryzed !
Could you add some tests for this and resolve the comments so we can put this in?

**Comment by jnoortheen** [comment]
Thanks for creating this. The history-alias now use functions from `cli_utils.py` . I've create another PR implementing this https://github.com/xonsh/xonsh/pull/4604

**Pull request closed without merging**